Library/Homebrew/dependable.rb

```diff
@@ -31,6 +31,10 @@ def required?
     !build? && !test? && !optional? && !recommended?
```

Comment:
```suggestion
    !build? && !test? && !fetch? && !optional? && !recommended?
```

Will also need to fix all the other places that `.test?` is used:

```
m1kebook # rg '\.test\?'                              (master) [~/OSS/Homebrew]
Library/Homebrew/cmd/deps.rb
165:      str = "#{str} [test]" if dep.test?

Library/Homebrew/language/python.rb
139:          next if d.build? || d.test?

Library/Homebrew/os/mac.rb
119:        next false if check_only_runtime_requirements && req.build? && !req.test?

Library/Homebrew/build.rb
55:      if req.prune_from_option?(build) || req.prune_if_build_and_not_dependent?(dependent, formula) || req.test?
66:         (dep.test? && !dep.build?)

Library/Homebrew/dev-cmd/test.rb
67:        next if dependency.test?

Library/Homebrew/dev-cmd/linkage.rb
48:        if args.test?

Library/Homebrew/dependencies_helpers.rb
47:      elsif dep.build? || dep.test?
49:        keep ||= dep.test? && includes.include?("test?") && dependent == root_dependent

Library/Homebrew/formula_installer.rb
539:        keep_build_test ||= req.test? && include_test? && dependent == f
544:           ((req.build? || req.test?) && !keep_build_test) ||
570:      keep_build_test ||= dep.test? && include_test? && @include_test_formulae.include?(dependent.full_name)
574:      if dep.prune_from_option?(build) || ((dep.build? || dep.test?) && !keep_build_test)

Library/Homebrew/test/dependency_expansion_spec.rb
113:      described_class.keep_but_prune_recursive_deps if dep.test?
```

Reply:
It's a little bit "weird" that I have to check all of these (not your fault, the code is as it-is, just some thoughts for the future). I am introducing a new attribute, that's all. I'm fine with the missing `fetch` in the class annotation method, and the missing fetch in the `required?` method. There you are right, they are necessary and I forgot to add them.

But for the others, it feels like a weird design: ideally we should migrate all these boolean logic checks to their respective classes, or refactor code so that a change/check does not feel necessary. Because else it means I can introduce code that will break other code in brew in unexpected ways, and this is bad (if you had not told me, I would have never checked all this code).

I made a few changes by adding fetch when necessary. Can you double check? I am not confortable with the `dependencies_helpers` and `formula_installer` part.

Reply:
> It's a little bit "weird" that I have to check all of these (not your fault, the code is as it-is, just some thoughts for the future). I am introducing a new attribute, that's all.

I disagree: you're adding a dependency that you don't actually want to be treated as a dependency in most cases 😁. It makes sense to me that it would need to be special-cased. Also, depending on the case, it may need to behave more like `:build` or `:fetch`.



> Because else it means I can introduce code that will break other code in brew in unexpected ways, and this is bad (if you had not told me, I would have never checked all this code).

I don't think it would break anything, it'll just result in a dependency that's added when it's not necessary.



> I made a few changes by adding fetch when necessary. Can you double check? I am not confortable with the `dependencies_helpers` and `formula_installer` part.

Will take a look!